numpy/array_api/_array_object.py

```diff
@@ -992,6 +1013,12 @@ def dtype(self) -> Dtype:
     def device(self) -> Device:
         return "cpu"
 
+    # Note: mT is new in array API spec (see matrix_transpose)
+    @property
+    def mT(self) -> Array:
```

Comment:
The new APIs for matrix transpose were chosen so that they can be as backwards compatible as possible with libraries like NumPy that already use .T for reversing the axes. See https://github.com/data-apis/array-api/issues/228 and the corresponding PRs.

Reply:
> I feel this was discussed in NumPy before (including the potential `.T` limitation!), time to revive that discussion!? 

Yes I think so.

> I find it mildly annoying that we do these things in the new namespace, but seem avers to discussing it for NumPy proper.

I don't think we are, it's just that no one followed up/through on making `matrix_transpose` public, and the discussion was framed around a big backcompat break in `.T`, which did not help. I think no one would object to making the private `matrix_transpose` public, it just needs doing?

> If we can get away with doing these things in NumPy it would be the best way to move code towards the better standard!

agreed